[LavanyaGarg] iP

lavanyagarg112 opened the pull request:
# Schedulo - Task Management Application

> "The key is not to prioritize what's on your schedule, but to schedule your priorities." – Stephen Covey

Schedulo is your personal assistant that helps you manage tasks efficiently. With a simple GUI, it allows you to keep track of your to-dos, deadlines, events, and more.

## Features

- **Task Management**
- **Deadline Tracking**
- **Event Scheduling**
- **Add multiple todos at once**
- **Task Searching**
- **Easy-to-use Graphic User Interface**

## How to Get Started

1. **Download** the application from [here](#).
2. **Launch** the application by double-clicking it.
3. **Add** your tasks, deadlines, and events.
4. **Enjoy** seamless task management with Schedulo! :smiley:

## Example Usage

Here's how you can get started with Schedulo:

```java
public class Schedulo {
    public static void main(String[] args) {
        Application.launch(App.class, args);
    }
}
```
## Example Command 

To add a new task in Schedulo, simply type:

`todo Read a book`

This will create a new to-do task with the description "Read a book."

## Upcoming Features

- [ ] **Reminders** - Get reminded of your tasks and deadlines.
- [ ] **Recurring Tasks** - Support for tasks that repeat on a schedule.
- [ ] **Task Prioritization** - Mark tasks as high or low priority.
- [ ] **UI Overhaul** - A new and improved graphical interface.

## Why Use Schedulo?

- It's **fast** and **responsive**.
- It's **simple** to use.
- It helps you **stay organized**.

And best of all, it's **FREE**!


Thank you for using Schedulo! We hope it makes managing your tasks easier. If you encounter any issues, please [submit an issue](#).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

lavanyagarg112 committed:
chatbot named - level 0 done

lavanyagarg112 committed:
chatbot named - level 0 done

lavanyagarg112 committed:
level 0 done, it echoes what we say

lavanyagarg112 committed:
add, list added, level 2 done

lavanyagarg112 committed:
classes made level 3 done

lavanyagarg112 committed:
different types of tasks added, level 4 done

lavanyagarg112 committed:
input checking added

lavanyagarg112 committed:
some checking added, not for all though

lavanyagarg112 committed:
error handling added

lavanyagarg112 committed:
delete added, level 6 done

lavanyagarg112 committed:
todo for self reference added as comments

lavanyagarg112 committed:
exception classes made

lavanyagarg112 committed:
exceptions added. level 5 done

lavanyagarg112 committed:
my todo for this ip has been modified

lavanyagarg112 committed:
tests added and new error handling added

lavanyagarg112 committed:
gitignore updated

lavanyagarg112 committed:
gitignore changed

lavanyagarg112 committed:
enums added

lavanyagarg112 committed:
unnecessary files removed

lavanyagarg112 committed:
todos added

lavanyagarg112 committed:
unnecessary folder

lavanyagarg112 committed:
gitignore

lavanyagarg112 committed:
file load create, need to now create save

lavanyagarg112 committed:
saving on bye

lavanyagarg112 committed:
saved after every operation

lavanyagarg112 committed:
Merge pull request #1 from lavanyagarg112/branch-Level-7

Branch level 7

lavanyagarg112 committed:
chatbot understands the date, need to teach time

lavanyagarg112 committed:
chatbot recognizes time

lavanyagarg112 committed:
Merge pull request #2 from lavanyagarg112/branch-Level-8

Branch level 8

lavanyagarg112 committed:
more oop added with more classes

lavanyagarg112 committed:
Merge pull request #3 from lavanyagarg112/branch-A-MoreOOP

more oop added with more classes

lavanyagarg112 committed:
packages made

lavanyagarg112 committed:
Merge pull request #4 from lavanyagarg112/branch-A-Packages

packages made

lavanyagarg112 committed:
Merge pull request #5 from lavanyagarg112/add-gradle-support

Add gradle support

lavanyagarg112 committed:
gradle configuration done

lavanyagarg112 committed:
junit testing with gradle done

lavanyagarg112 committed:
Merge pull request #6 from lavanyagarg112/branch-A-JUnit

junit testing with gradle done

lavanyagarg112 committed:
jar file creation done

lavanyagarg112 committed:
Merge branch 'master' of github.com:lavanyagarg112/ip

lavanyagarg112 committed:
java docs added for all

lavanyagarg112 committed:
coding standards followed for all

lavanyagarg112 committed:
find command added

lavanyagarg112 committed:
Merge pull request #7 from lavanyagarg112/branch-A-JavaDoc

java docs added for all

lavanyagarg112 committed:
merge conflicts resolved

lavanyagarg112 committed:
unnecessary imports removed

lavanyagarg112 committed:
Merge pull request #8 from lavanyagarg112/branch-A-CodingStandard

coding standards followed for all

lavanyagarg112 committed:
merged

lavanyagarg112 committed:
Merge pull request #9 from lavanyagarg112/branch-Level-9

Branch level 9

lavanyagarg112 committed:
readme updated

lavanyagarg112 committed:
coding standard made better

lavanyagarg112 committed:
coding standard made better by package

lavanyagarg112 committed:
java doc added for tests

lavanyagarg112 committed:
checktyle in progrss

lavanyagarg112 committed:
modified file to follow checkstyle

lavanyagarg112 committed:
modify file to follow checkstyle

lavanyagarg112 committed:
modify file to follow checkstyle

lavanyagarg112 committed:
modify file to follow checkstyle

lavanyagarg112 committed:
modify file to follow checkstyle

lavanyagarg112 committed:
modify file to follow checkstyle

lavanyagarg112 committed:
add exception handling in a better way

lavanyagarg112 committed:
basic fxml

Ansel-Ch commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,41 @@
+
+import task.TaskList;
+import task.Task;
+import exception.ScheduloException;
```
Perhaps this is in the wrong lexicographical order? 🤔

lavanyagarg112 replied:
Oh thanks for pointing it out, I will check the ordering

Ansel-Ch commented on src/main/java/command/MarkCommand.java:
```diff
@@ -0,0 +1,46 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws ScheduloException, IOException {
+        try {
+            tasks.mark(this.index - 1);;
```
It might be good to remove the extra semicolon.

lavanyagarg112 replied:
oh oops, thanks for point it out!

Ansel-Ch reviewed (commented):
LGTM! Just small mistakes here and there.

lavanyagarg112 committed:
gui created

lavanyagarg112 committed:
styles added

lavanyagarg112 committed:
Merge pull request #10 from lavanyagarg112/branch-Level-10

Branch level 10

lavanyagarg112 committed:
all duke instances removed

lavanyagarg112 committed:
modify as per style

lavanyagarg112 committed:
modify as per style

lavanyagarg112 committed:
modify as per style

lavanyagarg112 committed:
modify as per style

lavanyagarg112 committed:
modify as per style

lavanyagarg112 committed:
remove extra semicolon

lavanyagarg112 committed:
update docs

lavanyagarg112 committed:
use varargs for getting multiple todos at once

tsulemornorfi commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,41 @@
+     * Executes the AddCommand by adding the task to the task list and saving the updated list to storage.
+     *
+     * @param tasks   The TaskList to which the task will be added.
+     * @param ui      The Ui instance used to interact with the user.
```
The @param tag for 'ui' is unused.

tsulemornorfi commented on src/main/java/command/AddMultipleTodosCommand.java:
```diff
@@ -0,0 +1,29 @@
+import task.Todo;
+import util.Storage;
+
+public class AddMultipleTodosCommand extends Command {
```
Missing Javadoc comment

tsulemornorfi commented on src/main/java/command/AddMultipleTodosCommand.java:
```diff
@@ -0,0 +1,29 @@
+        storage.save(tasks);
+        return message;
+    }
+    
```
Theres a trailing whitespace.

tsulemornorfi commented on src/main/java/command/CommandType.java:
```diff
@@ -0,0 +1,63 @@
+package command;
+
+/**
+ * The CommandType enum defines the different types of commands that can be recognized and executed in the Schedulo application.
```
Line is longer than 120 characters

tsulemornorfi commented on src/main/java/schedulo/DialogBox.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    private void changeDialogStyle(Command commandType) {
+        if (commandType instanceof AddCommand){
```
A whitespace should precede the "{"

currynia reviewed (commented):
Looks mostly alright. Just some slight formatting issues :)

lavanyagarg112 committed:
multiple todos command

lavanyagarg112 committed:
add assertions for important assumptions

Add assert feature to document important assumptions that should hold at various points in the code.

lavanyagarg112 committed:
add usage of streams

Use of streams makes finding the keyword much more easy and readable

lavanyagarg112 committed:
add yml file for github actions

Add yml file for github actions to set up continuous integration

lavanyagarg112 committed:
workflow modified

lavanyagarg112 committed:
modify workflow for yml

lavanyagarg112 committed:
modify tests to follow the changes

modify tests to follow the new gui, as well as better test the system

lavanyagarg112 committed:
modify code to follow checkstyle better

lavanyagarg112 committed:
modify code to follow checkstyle guidelines

lavanyagarg112 committed:
Merge pull request #13 from lavanyagarg112/branch-A-CodeQuality

modify code to follow checkstyle guidelines

lavanyagarg112 committed:
update readme with minor change

lavanyagarg112 committed:
modfy boolean name to follow code quality

lavanyagarg112 committed:
Merge pull request #14 from lavanyagarg112/branch-A-CodeQuality

modfy boolean name to follow code quality

lavanyagarg112 committed:
modify javadoc for more description

lavanyagarg112 committed:
modify tests as per new system

lavanyagarg112 committed:
modify javadoc to add more details

lavanyagarg112 committed:
modify javadoc to add more details

lavanyagarg112 committed:
modify javadoc to add more details

lavanyagarg112 committed:
add help command to help user see all available commands

lavanyagarg112 committed:
modify code to follow checkstyle

modify code to follow checkstyle which states that the plus symbol should be on a new line

lavanyagarg112 committed:
modify code to follow coding style

lavanyagarg112 committed:
modify readme and help command to include all commands

lavanyagarg112 committed:
add ui.png to add the ui screenshot

lavanyagarg112 committed:
modify gui to make more engaging

lavanyagarg112 committed:
modify userguide to add details about the help command

lavanyagarg112 committed:
modify the response to give a unique personality

lavanyagarg112 committed:
add tests and modify checkstyle

lavanyagarg112 committed:
data folder created

lavanyagarg112 committed:
data folder added

lavanyagarg112 committed:
gitignore changed

lavanyagarg112 committed:
Modify storage to make new directory

lavanyagarg112 committed:
Update Readme to become concise

lavanyagarg112 committed:
Update Readme to add spacing

lavanyagarg112 committed:
Update Readme to add spacing

lavanyagarg112 committed:
Update Readme to add spacing

lavanyagarg112 committed:
Update Readme to add spacing

lavanyagarg112 committed:
Update Readme to add spacing

lavanyagarg112 committed:
Update Readme to remove spacing

lavanyagarg112 committed:
Modify storage file to remove trailing whitespace

lavanyagarg112 committed:
Modify the behaviour of bye to exit application

lavanyagarg112 committed:
Modify find command to be case insensitive
